### PR TITLE
Reinstate broken aps behaviour

### LINKED ISSUE #377: 1.7.5 does not accept aps payload in Notification constructor
Previously we could create a Notification with a payload like so:

    var notification = new apn.Notification({ aps: { badge: 1, message: "Hi there!" }})

However, b0bd4e7ba12d1fd0d4b841db6a3ada9232968a55 broke this behavior.

I'd be happy to contribute to fixing this issue, though it seems the easiest might be to simply revert that commit?

### PR DESCRIPTION
**Pull request by argon**
Fixes #377

### CODE REVIEW

File: lib/notification.js [hunk 1]
```diff
@@ -480,15 +480,15 @@ Notification.prototype.truncateStringToLength = function (string, length, encodi
...
 	if (this.contentAvailable) {
 		aps["content-available"] = 1;
 	}
 	
-	aps["url-args"] = this.urlArgs;
-	aps.category = this.category;
+	aps["url-args"] = this.urlArgs || aps["url-args"];
+	aps.category = this.category || aps.category;
 
 	return Object.keys(aps).reduce(function(populated, key) {
```
Comment (agnoster) [inline lib/notification.js]: This is super duper minor, but just occurred to me: why use `reduce` over `find`? Seems like it'd be clearer?

```
return Object.keys(aps).find(function(key) {
  return aps[key] !== undefined;
}) ? aps : undefined;
```

(Technically also means it'll short-circuit as soon as it finds a populated key, but that doesn't seem like it's likely to be a meaningful performance win given the size of the objects ;-))

Comment (argon) [inline lib/notification.js]: Didn't think to use `find`. I'll change it now

**Comment by argon** [comment]
@agnoster any update on this?

**Comment by agnoster** [comment]
Unfortunately this isn't as high a priority for us right now because (due to circumstances that have absolutely nothing to do with the node-apn module) we're rewriting our push notification delivery system as a new service in Scala.

That said, the code looks like it fixes the issue and there's a decent chance we're not the only ones who were using it that way so it seems worth merging.

**Comment by A1bi** [comment]
Would you please create a new release that includes this fix? This bug just cost me several hours over the past two days after upgrading from an older version preventing all notifications from showing up on devices. I can confirm that the code on master works fine again.

**Comment by argon** [comment]
I apologise for that, I thought I had already released it. I'll do that now

**Comment by A1bi** [comment]
Wow, that's a quick response. Thanks a lot! :blush:

**Comment by argon** [comment]
Please try now, 1.7.6 has been published on npm

**Comment by A1bi** [comment]
Works like a charm, thanks again!

**Pull request merged**